test/e2e_node/eviction_test.go

```diff
@@ -103,6 +104,53 @@ var _ = SIGDescribe("InodeEviction", framework.WithSlow(), framework.WithSerial(
 	})
 })
 
+// TerminatedPodsEvictionOnDiskPressure tests that the node does not evict pods
+// when there are terminated pods and the image garbage collection can remove
+// their resources without evicting running pods
+var _ = SIGDescribe("TerminatedPodsEvictionOnDiskPressure", framework.WithSlow(), framework.WithSerial(), framework.WithDisruptive(), nodefeature.Eviction, func() {
+	f := framework.NewDefaultFramework("terminated-pods-eviction-on-disk-pressure-test")
+	f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
+	pressureTimeout := 10 * time.Minute
+	expectedNodeCondition := v1.NodeDiskPressure
+	expectedStarvedResource := v1.ResourceEphemeralStorage
+
+	ginkgo.Context(fmt.Sprintf(testContextFmt, expectedNodeCondition), func() {
+		tempSetCurrentKubeletConfig(f, func(ctx context.Context, initialConfig *kubeletconfig.KubeletConfiguration) {
+			// Calculate eviction thresholds based on available disk space
+			summary := eventuallyGetSummary(ctx)
+			availableBytesOnSystem := *(summary.Node.Fs.AvailableBytes)
+
+			diskConsumedByTest := resource.MustParse("4Gi")
+			evictionThreshold := strconv.FormatUint(availableBytesOnSystem-uint64(diskConsumedByTest.Value()), 10)
+
+			if availableBytesOnSystem <= uint64(diskConsumedByTest.Value()) {
+				e2eskipper.Skipf("Not enough disk space to run the test, available: %d", availableBytesOnSystem)
+			}
+
+			initialConfig.EvictionHard = map[string]string{string(evictionapi.SignalNodeFsAvailable): evictionThreshold}
+			initialConfig.EvictionMinimumReclaim = map[string]string{}
+		})
+
+		runEvictionTest(f, pressureTimeout, expectedNodeCondition, expectedStarvedResource, logDiskMetrics, []podEvictSpec{
+			{
+				evictionPriority: 0,
+				pod:              innocentPod(),
+			},
+			{
+				evictionPriority: 0,
+				pod:              diskConsumingPod("container-disk-hog", lotsOfDisk, &v1.VolumeSource{EmptyDir: &v1.EmptyDirVolumeSource{}}, v1.ResourceRequirements{}),
+			},
+			// This is the terminated pod, runEvictionTest will wait for it to finish
+			{
+				evictionPriority: 1,
+				// This is probably working because empty dir is being cleared when the
```

Comment:
This probably means that it is not testing what we want?

Maybe we should replicate what we do in `ImageGCNoEviction` test below.

Create 2 pods:
1. running pod with 110000 files
2. terminated pod with 110000 files in rootfs (not emptyDir)

only the second pod should be evicted due to exceeding inodes threshold.

Reply:
I added a new test with the following config:
- running pod with 80,000 inods
- terminated pod with 30,000 inods in rootfs (not emptyDir)

This causes the eviction threshold (100,000 inodes) to be passed with a total of 110,000 inodes, making the node to react evicting all terminated pods, freeing the 30,000 inodes written by the terminated pod and eliminating the disk pressure without affecting the running pod